### PR TITLE
Agenda sort

### LINKED ISSUE #128: Configure group based sorting in agenda
### Does this feature exist in Emacs orgmode core?

N/A

### Orgmode link

_No response_

### Feature value

The sorting of tasks in the agenda view seems a bit arbitrary - I get a mix of NEXT and TODO items, and the files they come from are also mixed even if all the tasks are for the same date. I couldn't find how sorting is done in the help file...

For e.g. I get:
```python
понеделник 01 ноември 2021
  work:       Scheduled:  NEXT task name
  nehm:       Scheduled:  NEXT task name
  nehm:       Scheduled:  NEXT task name
  pros:       Scheduled:  NEXT task name
  work:       Scheduled:  NEXT task name
  nehm:       Scheduled:  NEXT task name
  nehm:       Scheduled:  TODO task name
  evoc:       Scheduled:  TODO task name
  nehm:       Scheduled:  NEXT task name
```

It would be very helpful if there is a way to group items based on keywords and files, and possibly set different priorities for sorting them in the agenda, e.g. `DONE > DOING > NEXT > TODO` :+1: 

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by gerazov**
Closes #128

### CODE REVIEW

**Comment by gerazov** [comment]
Please squash the commits :sweat_smile:

**Comment by gerazov** [comment]
ok now not same day tasks follow todays tasks that follow time of day tasks, I'm trying to sort the tasks now

File: lua/orgmode/agenda/init.lua [hunk 1]
```diff
@@ -457,16 +472,27 @@ function Agenda:agenda()
...
+          table.insert(date.agenda_items, agenda_item)
+        end
       end
     end
-
     date.agenda_items = sort_agenda_items(date.agenda_items)
+    date_not_same.agenda_items = sort_agenda_items(date_not_same.agenda_items)
+    date_only.agenda_items = sort_agenda_items_categories(date_only.agenda_items, category_inds)
 
+    date.agenda_items = utils.concat(date.agenda_items, date_only.agenda_items, false)
```
Comment (kristijanhusak) [inline lua/orgmode/agenda/init.lua:494]: @gerazov lets go with some solution to do all this sorting in the `sort_agenda_items` function. Saving all this additional state can potentially slow down agenda rendering, especially on month/year view.

Comment (gerazov) [reply lua/orgmode/agenda/init.lua:494]: Ok I'll refactor :sweat_smile:

File: lua/orgmode/utils/init.lua [hunk 2]
```diff
@@ -427,4 +427,31 @@ function utils.promisify(fn)
   return fn
 end
 
+---debug printing a table
+---source https://stackoverflow.com/questions/41942289/display-contents-of-tables-in-lua
+function utils.tprint(tbl, indent)
```
Comment (kristijanhusak) [inline lua/orgmode/utils/init.lua:432]: Remove this and use `vim.inspect(tbl)` instead.

**Comment by kristijanhusak** [comment]
@gerazov please squash the commits into a single one, and clean up the commented and unused code.

**Comment by kristijanhusak** [comment]
Closing this in favor of https://github.com/nvim-orgmode/orgmode/commit/ed8a420e8437e7074c94cf4c321eaaf22b055aa7

**Pull request closed without merging**